src/primitive.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2018 King's College London created by the Software Development
+// Team <http://soft-dev.org/>
+//
+// The Universal Permissive License (UPL), Version 1.0
+//
+// Subject to the condition set forth below, permission is hereby granted to any
+// person obtaining a copy of this software, associated documentation and/or
+// data (collectively the "Software"), free of charge and under any and all
+// copyright rights in the Software, and any and all patent rights owned or
+// freely licensable by each licensor hereunder covering either (i) the
+// unmodified Software as contributed to or provided by such licensor, or (ii)
+// the Larger Works (as defined below), to deal in both
+//
+// (a) the Software, and (b) any piece of software and/or hardware listed in the
+// lrgrwrks.txt file if one is included with the Software (each a "Larger Work"
+// to which the Software is contributed by such licensors),
+//
+// without restriction, including without limitation the rights to copy, create
+// derivative works of, display, perform, and distribute the Software and make,
+// use, sell, offer for sale, import, export, have made, and have sold the
+// Software and the Larger Work(s), and to sublicense the foregoing rights on
+// either these or other terms.
+//
+// This license is subject to the following condition: The above copyright
+// notice and either this complete permission notice or at a minimum a reference
+// to the UPL must be included in all copies or substantial portions of the
+// Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+use byteorder::{ByteOrder, LittleEndian};
+
+use rustc_const_math::ConstInt;
+use rustc::mir::{Operand, Constant, Literal};
+use rustc::middle::const_val::ConstVal;
+use rustc::ty::{Ty, TypeVariants};
+use syntax::ast::{IntTy, UintTy};
+
+use interp::TypedVal;
+use machine::Machine;
+
+/// An operand's value can always be determined without needing to temporarily
+/// save values to a stack. This is because of the recursive nature of the MIR.
+/// An operand will only ever return a primitive value or a pointer.
+pub fn eval_operand<'a,'tcx>(operand: &Operand<'tcx>,
+                             vm: &mut Machine<'a, 'tcx>)
+                             -> TypedVal<'tcx> {
+    let ty = operand.ty(vm.cur_frame().mir, vm.tcx);
+    match *operand {
+        Operand::Copy(ref place) |
+        Operand::Move(ref place) => {
```

Comment:
Very confusing formatting? Why not put `Operand::Move(ref place)` on the line above?

Reply:
I could do that. This was the way it's formatted as part of the visitor pattern within rustc. So I assumed it was idiomatic.

Reply:
Hmm. I don't particularly like it, but if there's a good precedent for it, let's stick with it.

Reply:
I'm not sure I do either. It does have the drawback of being misread at a glance. From what I've gathered reading rustc's source code, the code styling is quite focused on reducing line length. You can see this in most of the method signatures, even if they would comfortably fit onto one line.